src/sign.ml

```diff
@@ -0,0 +1,82 @@
+open Cmdliner
+
+let sign days is_ca client altname key cacert csr certfile =
+  match Common.(read_pem key, read_pem cacert, read_pem csr) with
+  | Common.Ok key, Common.Ok cacert, Common.Ok csr ->
+     let key = X509.Encoding.Pem.Private_key.of_pem_cstruct1 key
+     and cacert = X509.Encoding.Pem.Certificate.of_pem_cstruct1 cacert
+     and csr = X509.Encoding.Pem.Certificate_signing_request.of_pem_cstruct1 csr
+     in
+     let ent = match is_ca, client with
+       | true, _ -> `CA
+       | false, true -> `Client
+       | false, false -> `Server
+     in
+     let name =
+       if altname then
+         let info = X509.CA.info csr in
+         match List.filter (function `CN _ -> true | _ -> false) info.X509.CA.subject with
+         | [ `CN x ] -> Some x
+         | _ -> None
+       else
+         None
+     in
+     let issuer = X509.subject cacert in
+     let pubkey = X509.public_key cacert in
+     Nocrypto_entropy_unix.initialize ();
+     (match Common.sign days key pubkey issuer csr name ent with
+      | Common.Error str -> Printf.eprintf "%s\n" str; `Error
+      | Common.Ok cert ->
+         (match Common.write_pem certfile (X509.Encoding.Pem.Certificate.to_pem_cstruct1 cert) with
+          | Common.Ok () -> `Ok
+          | Common.Error str -> Printf.eprintf "%s\n" str; `Error))
+  | Common.Error str, _, _
+  | _, Common.Error str, _
+  | _, _, Common.Error str -> Printf.eprintf "%s\n" str; `Error
+
+let client =
+  let doc = "Client or server certificate" in
+  Arg.(value & flag & info ["client"] ~doc)
+
+let altname =
+  let doc = "include a subjectAltName in certificate" in
+  Arg.(value & flag & info ["altname"] ~doc)
```

Comment:
Can you be more specific in the docstring about what SAN will be set, since it's not user-specified?